libs/ImpExplorer.class.lib.nut

```diff
@@ -0,0 +1,515 @@
+                if (config.remoteTrigger != null) {
+                    local rp = RemotePin(config.remoteTrigger.pin);
+                    rp.configure(DIGITAL_OUT, config.remoteTrigger.state);
+                    rp.write(config.remoteTrigger.state, config.remoteTrigger.duration, decrementProcesses.bindenv(this));
+                }
+                break;
+        }
+
+    }
+    
+
+    // function that posts the readings and sends the device to sleep
+    // 
+    // @param     none
+    // @returns   none
+    // 
+    function postReadings() {
+        globalDebug.log("postingReadings");
+        // RSSI doesn't get a value when offline, add it now if required
+        if (reading.rssi == 0) reading.rssi = imp.getrssi();
+        // Add the location after a blinkup or new squirrel
+        if (hardware.wakereason() == WAKEREASON_BLINKUP || hardware.wakereason() == WAKEREASON_NEW_SQUIRREL) {
+            reading._location <- imp.scanwifinetworks();
+        }
+        
+
+
+        // Send the reading
+        if (config.C1mode == 2) {
+            lora.sendMessage(buildLoRaPacket());
+        } else {
+            //conctr.sendData(reading);
+            agent.send("reading", reading);
+        }
+
+        // Reset and save timeout counter
+        ::nv.sleepTime = NO_WIFI_SLEEP_PERIOD;
+
+        
+        // Determine how long to wait before sleeping
+        local sleepdelay = 20;
+        if (hardware.wakereason() == WAKEREASON_TIMER) {
+            sleepdelay = 0.5;
+        } else if (hardware.wakereason() == WAKEREASON_PIN) {
+            sleepdelay = 5;
+            agent.send("config", true);
+        }
+
+        // Checks if stayAwake is enabled.
+        if (config.stayAwake) {
+            imp.setpowersave(true);
+            //wakepin.configure(DIGITAL_IN, poll.bindenv(this));
+            if (config.measurementCycles == null) {
+                imp.wakeup(2, poll.bindenv(this));
+            } else {
+                imp.wakeup(config.measurementCycles, poll.bindenv(this));    
+            }
+        } else {
+            // Wait the specified time
+            server.flush(10);
+            imp.wakeup(sleepdelay, function() {
+
+                // Determine how long to sleep for
+                local sleepTime = calcSleepTime(reading.battery, hardwareType);
+
+                // Now actually sleep
+                wakepin.configure(DIGITAL_IN_WAKEUP);
+                server.sleepfor(sleepTime); 
+                
+            }.bindenv(this));
+        }
+        
+    }
+    
+    // function that builds a packet to be sent via LoRaWAN
+    // 
+    // @param     none
+    // @returns   messageBlob - 11 byte packet containing the sensor readings
+    // 
+    function buildLoRaPacket() {
+        local messageBlob = blob(11);
+        local tempBlob = blob(2);
+        local presBlob = blob(4);
+        local humidBlob = blob(4);
+        
+        tempBlob.writen((reading.temperature*100.0),'s');
```

Comment:
Here and in a few other places in this function you're writing floats to the blob as integers. Just checking that they're all intentional.